clients/src/main/java/org/apache/kafka/common/requests/ElectLeadersRequest.java

```diff
@@ -75,6 +76,13 @@ private ElectLeadersRequestData toRequestData(short version) {
                         data.topicPartitions().add(tps);
                     }
                     tps.partitions().add(tp.partition());
+                    if (version >= 3) {
```

Comment:
Honestly I find it annoying that this class does this kind of "manual serialization" rather than simply using the generated KRPC classes. Anyway, if we start using a subclass of TopicPartition, then this will change.